pkg/acquisition/modules/docker/docker.go

```diff
@@ -0,0 +1,452 @@
+			return &ContainerConfig{ID: container.ID, Name: container.Names[0], Labels: d.Config.Labels}, true
+		}
+	}
+
+	for _, containerName := range d.Config.ContainerName {
+		for _, name := range container.Names {
+			if strings.HasPrefix(name, "/") && len(name) > 0 {
+				name = name[1:]
+			}
+			if name == containerName {
+				return &ContainerConfig{ID: container.ID, Name: name, Labels: d.Config.Labels}, true
+			}
+		}
+
+	}
+
+	for _, cont := range d.compiledContainerID {
+		if matched := cont.Match([]byte(container.ID)); matched {
+			return &ContainerConfig{ID: container.ID, Name: container.Names[0], Labels: d.Config.Labels}, true
+		}
+	}
+
+	for _, cont := range d.compiledContainerName {
+		for _, name := range container.Names {
+			if matched := cont.Match([]byte(name)); matched {
+				return &ContainerConfig{ID: container.ID, Name: name, Labels: d.Config.Labels}, true
+			}
+		}
+
+	}
+
+	return &ContainerConfig{}, false
+}
+
+func (d *DockerSource) WatchContainer(monitChan chan *ContainerConfig) error {
+	ticker := time.NewTicker(1 * time.Second)
+
+	for {
+		select {
+		case <-d.t.Dying():
+			d.logger.Infof("stopping group watch")
+			return nil
+		case <-ticker.C:
+			runningContainer, err := d.Client.ContainerList(context.Background(), dockerTypes.ContainerListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, container := range runningContainer {
+				// don't need to re eval an already monitored container
+				if _, ok := d.runningContainerState[container.ID]; ok {
+					continue
+				}
+				if containerConfig, ok := d.EvalContainer(container); ok {
+					monitChan <- containerConfig
+				}
+			}
+			ticker.Reset(1 * time.Second)
+		}
+	}
+}
+
+func (d *DockerSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	d.t = t
+	monitChan := make(chan *ContainerConfig)
+	d.logger.Infof("Starting docker acquisition")
+	t.Go(func() error {
+		return d.DockerManager(monitChan, out)
+	})
+
+	return d.WatchContainer(monitChan)
+}
+
+func (d *DockerSource) Dump() interface{} {
+	return d
+}
+
+func ReadTailScanner(scanner *bufio.Scanner, out chan string, t *tomb.Tomb) error {
+	for scanner.Scan() {
+		out <- scanner.Text()
+	}
+	return nil
+}
+
+func (d *DockerSource) TailDocker(container *ContainerConfig, outChan chan types.Event) error {
+	container.logger.Infof("start tail for container %s", container.Name)
+	dockerReader, err := d.Client.ContainerLogs(context.Background(), container.ID, *d.containerLogsOptions)
+	if err != nil {
+		container.logger.Errorf("unable to read logs from container: %+v", err)
+		return err
+	}
+	// we use this library to normalize docker API logs (cf. https://ahmet.im/blog/docker-logs-api-binary-format-explained/)
+	reader := dlog.NewReader(dockerReader)
+	scanner := bufio.NewScanner(reader)
+	readerChan := make(chan string)
+	readerTomb := &tomb.Tomb{}
+	readerTomb.Go(func() error {
+		return ReadTailScanner(scanner, readerChan, readerTomb)
+	})
+	for {
```

Comment:
what happens if the docker container (and/or service) is dying while we're tailing it ?